docker-storage-setup.sh

```diff
@@ -537,6 +537,32 @@ is_block_dev_partition() {
   return 1
 }
 
+check_block_dev_sig() {
+  local bdev=$1
+  local sig
+
+  [ ! -b "$bdev" ] && bdev="/dev/${bdev}"
+
+  if ! sig=$(wipefs -p $bdev); then
+    Fatal "Failed to check signatures on device $bdev"
+  fi
+
+  [ "$sig" == "" ] && return 0
+
+  while IFS=, read offset uuid label type; do
+    [ "$offset" == "# offset" ] && continue
+
+    # Ignore dos signatures for now. We have already checked that disk
+    # does not have partitions. But there might be leftover partition
+    # signature and so far we have not forced user to wipe it. So in an
+    # attempt to not change user visible behavior, don't force user to
+    # wipe dos signature and just make sure disk is not partitioned.
```

Comment:
Ok, lets go with that. If there is lot of hue and cry, we can always add a patch to special case this. I was not very happy with special casing either.